crates/ide/src/ide/assists/remove_empty_inherit.rs

```diff
@@ -0,0 +1,72 @@
+//! Remove an empty `inherit`.
+//!
+//! ```nix
+//! { foo = "bar"; inherit; }
+//! ```
+//! =>
+//! ```nix
+//! { foo = "bar"; }
+//! ```
+use super::{AssistKind, AssistsCtx};
+use crate::TextEdit;
+use syntax::ast::{self, AstNode};
+use syntax::{TextRange, TextSize};
+
+pub(super) fn remove_empty_inherit(ctx: &mut AssistsCtx<'_>) -> Option<()> {
+    let node = ctx.covering_node::<ast::Inherit>()?;
+
+    // There should be no attrs inherited.
+    if node.attrs().next().is_some() {
+        return None;
+    }
+
+    let syntax = node.syntax();
+    let range = if matches!(syntax.last_token()?.next_token(),
+        Some(tok) if tok.text().starts_with(|c: char| c.is_whitespace()))
+    {
+        // Remove trailing whitespace.
+        let range = syntax.text_range();
+        TextRange::new(range.start(), range.end() + TextSize::from(1))
+    } else {
+        syntax.text_range()
+    };
```

Comment:
Why a magic number `1`? Whitespaces can be checked directly via `kind()` and you could get its `text_range` without a `starts_with` scan.

```suggestion
    let mut range = node.syntax().range();
    // Also remove trailing SPACEs.
    if let Some(ws) = node.syntax().last_token()?.next_token().filter(|tok| tok.kind() == SyntaxKind::SPACE) {
        range = range.cover(ws.text_range());
    }
```

Reply:
thanks! I didn't know that was possible